src/test/java/io/inholland/groep4/api/service/AccountsServiceTest.java

```diff
@@ -0,0 +1,159 @@
+    public void GettingAccountWithWrongIDShouldThrowException(){
+        Long idToFind = 100l;
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            userAccountService.getAccountById(idToFind);
+        });
+
+        assertTrue(exception.getMessage().contains("Id not found"));
+    }
+
+    @Test
+    public void gettingAccountByUserShouldGiveUserAccount(){
+        String usernameToFind = "test-user1";
+        User username = userService.findByUsername(usernameToFind);
+        List<UserAccount> userAccount = userAccountService.getAccountsByUser(username);
+
+        assertThat(userAccount).isNotEmpty();
+        assertThat(username.getUsername()).isEqualTo("test-user1");
+    }
+    @Test
+    public void AddIncorrectNewIBAN(){
+        User user = new User();
+        user.setUsername("peter");
+        user.setPassword("test");
+        user.setFirstName("Peter");
+        user.setLastName("Griffin");
+        user.setEmail("peter@example.com");
+        user.setBirthdate("01/01/1970");
+
+        UserAccount userAccount = new UserAccount();
+        userAccount.setAccountType(UserAccount.AccountTypeEnum.CURRENT);
+        userAccount.setOwner(user);
+        userAccount.setAccountBalance(0.00);
+        userAccount.setAccountStatus(UserAccount.AccountStatusEnum.ACTIVE);
+        userAccount.setLowerLimit(100.00);
+        userAccount.setIBAN("NL01INHO0000000001");
+
+        String IBAN = userAccountService.getIBAN();
+        boolean ExistByIBAN = userAccountService.existByIBAN(IBAN);
+        UserAccount newUser = userAccountService.add(userAccount, ExistByIBAN);
+
+        assertThat(newUser).isNotNull();
+    }
+
+    @Test
+    public void CheckIfAccountBelongToUser(){
+        String usernameToFind = "test-user1";
+        User username = userService.findByUsername(usernameToFind);
+        Long idToFind = 8l;
+
+        boolean account = userAccountService.checkIfAccountBelongsToOwner(username, idToFind);
+        assertThat(account).isEqualTo(true);
+    }
+
+    @Test
+    public void IfAccountDoesNotBelongToUserShouldThrowException(){
+        String usernameToFind = "test-user1";
+        User username = userService.findByUsername(usernameToFind);
+        Long idToFind = 100l;
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            userAccountService.checkIfAccountBelongsToOwner(username, idToFind);
+        });
+
+        assertFalse(exception.getMessage().contains("Account not found"));
+    }
+
+    @Test
+    public void savingUserAccount()
+    {
+        Long idToFind = 1l;
+        UserAccount account = userAccountService.getAccountById(idToFind);
+        UserAccount userAccount = userAccountService.save(account);
+
+        assertThat(userAccount.getAccountBalance()).isEqualTo(500.0);
+    }
+
+    @Test
+    public void gettingIBAN(){
+        String IBAN = userAccountService.getIBAN();
+
+        assertThat(IBAN).isNotNull();
```

Comment:
Deze moet uitgebreider, je kan bijvoorbeeld kijken of de IBAN met "NL" begint, en een bepaalde lengte heeft.